File: prototype/RenderBuffer.js
class RenderBuffer
{
    constructor(sys, gl, position, color, texture, vertexLimit, indexLimit)
    {
        this.vao;
        this.vbo;
        this.ebo;
        this.vertexPos = 0;
        this.indexPos = 0;
        this.indexOffset = 0;
        this.vertices = new Float32Array(vertexLimit * (position + color + texture));
        this.indices = new Uint16Array(indexLimit);
        RenderSystem.MakeVao(sys, gl, this, position, color, texture);
    }
    static Zero(buffer)
    {
        buffer.vertexPos = 0;
        buffer.indexPos = 0;
        buffer.indexOffset = 0;
    }
}

File: native/src/graphics/renderbuffer.c
#include "renderbuffer.h"

renderbuffer *renderbuffer_init(int position, int color, int texture, int vertex_limit, int index_limit) {
    renderbuffer *r = safe_malloc(sizeof(renderbuffer));
    r->position = position;
    r->color = color;
    r->texture = texture;
    r->vertex_pos = 0;
    r->vertex_limit = vertex_limit;
    r->index_pos = 0;
    r->index_limit = index_limit;
    r->index_offset = 0;
    r->vertices = safe_malloc(sizeof(GLfloat) * vertex_limit);
    r->indices = safe_malloc(sizeof(GLuint) * index_limit);
    return r;
}

void renderbuffer_zero(renderbuffer *self) {
    self->vertex_pos = 0;
    self->index_pos = 0;
    self->index_offset = 0;
}
